packages/idyll-ast/src/index.js

```diff
@@ -53,11 +62,30 @@ const getText = function(node) {
 
 const walkNodes = function(ast, f) {
   (ast || []).forEach((node) => {
-    walkNodes(getChildren(node), f);
+    walkNodes(getChildren(node), f); 
     f(node);
   });
 };
 
+/**
+ * function to do a depth first traversal on the ast tree. 
+ * @param {*} ast     Array that forms the tree structure 
+ * @param {*} f       call-back function 
+ */
+const dfsTraversal = function (ast, f) {
+  (ast || []).forEach((node) => {
+    if(typeof node !== 'string') {    
+      f(node); 
+      childAst = getChildren(node); 
+      (childAst || []).forEach((node) => {
+          f(node);
+      });
+      (childAst || []).forEach((node) => {
+          dfsTraversal(getChildren(node), f); 
+      });
+   }
```

Comment:
It seems like the callback would never fire for text nodes?

Reply:
Is there a way to avoid reaching down two levels (calling `getChildren` twice) here?  Notice the logic for `walkNodes` only goes down a single level.